ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -303,39 +333,61 @@ export const PageProviderDisplayBehavior = [
     },
 
     deselectItem(item) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllActive) {
         this.$.list.deselectItem(item);
         this._updateFlags();
       }
     },
 
     deselectIndex(index) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllActive) {
         this.$.list.deselectIndex(index);
         this._updateFlags();
       }
     },
 
     selectAll() {
-      this.items.forEach(
-        function(item) {
-          this.selectItem(item);
-        }.bind(this.$.list),
-      );
-      this._updateFlags();
+      if (this.selectionEnabled && this.selectAllEnabled) {
+        this._isSelectAllActive = true;
+        // select the visible items first to speed up the first paint (the others can be deferred)
+        const { start, end } = this._getSelectionBoundaries();
+        this._updateSelectedItems((index) => {
+          this.selectItem(this.items[index]);
+        });
+
+        // basic method to push items to selected items without proper selection, to speed up rendering
+        const pushSelectedItems = (indexStart, limit) => {
```

Comment:
Instead of always redefining this method when we do a select all, perhaps we could declare it as a private member of the behavior?

Reply:
Yes, done. Thanks